ocaml/xapi/xapi_pool_update.ml

```diff
@@ -499,7 +499,10 @@ let resync_host ~__context ~host =
          && Xapi_pool_patch.pool_patch_of_update ~__context self
             |> fun self -> Db.Pool_patch.get_host_patches ~__context ~self
             |> function [] -> false | _ -> true)
-    |> List.iter (fun self -> destroy ~__context ~self)
+    |> List.iter (fun self -> destroy ~__context ~self);
```

Comment:
Please would you put this change into a commit on its own?
This is the change which is intended to fix the bug, whereas the other code-changes are just refactoring with no change to behaviour.